### DISCUSSION TITLE
Automatically renewing the letsencrypt certificate without restarting OME?

### DISCUSSION THREAD

**Discussion by vampirefrog** [OvenMediaLabs/OvenMediaEngine#1468] [Help] [answered]
Hello, is it possible to renew a letsencrypt certificate (with certbot), without having to restart OME? I'm running the OME API on port 80. I am aware of the `reloadCertificates` API call, but that's not the issue. The issue is that certbot needs port 80 to be unbound when it runs, for the `http-01` challenge. The way I solved this is, I've modified `/lib/systemd/system/certbot.service` as you can see below, and whenever the certificate is renewed, which, granted, is not that often, it stops OME before renewing and starts it after renewing.

```
[Unit]
Description=Certbot
Documentation=file:///usr/share/doc/python-certbot-doc/html/index.html
Documentation=https://certbot.eff.org/docs
[Service]
Type=oneshot
ExecStart=/usr/bin/certbot -q renew --no-random-sleep-on-renew --pre-hook="service ovenmediaengine stop" --post-hook="service ovenmediaengine start"
PrivateTmp=true
```

I guess another solution is to not use port 80 for OME, but is there another solution? Thank you.

**Comment by bchah** [comment, marked as answer]
I think the best solution is to use another API port for OME. In my application I use PHP and cURL to relay all requests to the OME API at http://0.0.0.0:API-PORT. So the API can be exposed but still controlled (input validation, restrictions etc)

**Reply by vampirefrog** [reply]
Yeah that's what I ended up doing. Using port 8080 for OME and certbot can use port 80 temporarily for the `http-01` challenge for renewal. I actually don't even use the non-ssl API so I might even disable it, if that's possible.

**Reply by vampirefrog** [reply]
Then again, I do use the plain HTTP port to POST to :reloadCertificates so it looks like I can't really disable it.
